src/OpenLoco/src/GameCommands/CreateIndustry.cpp

```diff
@@ -0,0 +1,722 @@
+        S5::getOptions().madeAnyChanges = 1;
+        if (World::TileManager::wasRemoveOnLastElement())
+        {
+            return World::TileClearance::ClearFuncResult::allCollisionsRemoved;
+        }
+        return World::TileClearance::ClearFuncResult::collisionRemoved;
+    }
+
+    /* 0x004551CC
+       bh:7 = buildImmediately
+       bh:0-3 = rotation
+       edi = randColour << 16
+       dl = buildingType
+       dh = industryId
+       ax = pos.x
+       cx = pos.y
+       bl = flags
+    */
+    static currency32_t placeIndustryBuilding(const IndustryId industryId, const World::Pos2& pos, const uint8_t direction, const uint8_t buildingType, const Colour colour, const bool buildImmediate, const uint8_t flags)
+    {
+        auto* industry = IndustryManager::get(industryId);
+        auto* indObj = industry->getObject();
+        if (!World::TileManager::checkFreeElementsAndReorganise())
+        {
+            return FAILURE;
+        }
+
+        const bool isMultiTile = indObj->buildingSizeFlags & (1U << buildingType);
+
+        // Workout the max surface height for building footprint
+        const auto buildingFootprint = getBuildingTileOffsets(isMultiTile);
+
+        // 0x00E0C3CB (note this is smallZ)
+        World::SmallZ highestBaseZ = 0;
+        for (const auto& offset : buildingFootprint)
+        {
+            const auto tilePos = World::toTileSpace(pos + offset.pos);
+            if (!World::validCoords(tilePos))
+            {
+                continue;
+            }
+            auto tile = World::TileManager::get(tilePos);
+            const auto* surface = tile.surface();
+
+            auto baseZ = World::TileManager::getSurfaceCornerHeight(*surface);
+            highestBaseZ = std::max<World::SmallZ>(highestBaseZ, baseZ);
+            if (surface->water())
+            {
+                highestBaseZ = std::max<World::SmallZ>(highestBaseZ, surface->water() * World::kMicroToSmallZStep);
+            }
+        }
+
+        // Workout clearance height of building (including scaffolding if required)
+        const auto buildingParts = indObj->getBuildingParts(buildingType);
+        // 0x00E0C3BC (note this is bigZ and does not include the base height)
+        auto clearHeight = 0;
+        for (auto part : buildingParts)
+        {
+            clearHeight += indObj->buildingPartHeight[part];
+        }
+        if (!buildImmediate && indObj->scaffoldingSegmentType != 0xFF)
+        {
+            auto* scaffObj = ObjectManager::get<ScaffoldingObject>();
+            bool requiresMore = clearHeight % scaffObj->segmentHeights[indObj->scaffoldingSegmentType];
+            const auto numSegments = (clearHeight / scaffObj->segmentHeights[indObj->scaffoldingSegmentType]) + (requiresMore ? 1 : 0);
+            clearHeight = (numSegments * scaffObj->segmentHeights[indObj->scaffoldingSegmentType]) + scaffObj->roofHeights[indObj->scaffoldingSegmentType];
```

Comment:
The expression `scaffObj->segmentHeights[indObj->scaffoldingSegmentType]` is repeated quite a few times here. Could we extract it to a const temporary here?
```suggestion
            const auto segmentHeight = scaffObj->segmentHeights[indObj->scaffoldingSegmentType];
            bool requiresMore = clearHeight % segmentHeight;
            const auto numSegments = (clearHeight / segmentHeight) + (requiresMore ? 1 : 0);
            clearHeight = (numSegments * segmentHeight) + scaffObj->roofHeights[indObj->scaffoldingSegmentType];
```